frontend/src/slices/auth.ts

```diff
@@ -0,0 +1,104 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import type { RootState } from "../store/store";
+
+interface AuthError {
+  message: string;
+}
+
+interface AuthState {
+  isAuth: boolean;
+  isLoading: boolean;
+  currentUser?: CurrentUser;
+  error?: AuthError;
+}
+
+interface CurrentUser {
+  id: string;
+}
+
+const initialState: AuthState = {
+  isAuth: false,
+  isLoading: false,
+};
+
+export const register = createAsyncThunk(
+  "auth/register",
+  async ({ email, password }: { email: string; password: string }) => {
```

Comment:
suggestion: this `{email: string; password: string}` type could be extracted into an interface, since it is used in multiple places and will probably be used in some components as well.